src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers/IdentifierUpgradeAnalyzer.cs

```diff
@@ -31,46 +48,69 @@ public override void Initialize(AnalysisContext context)
             context.EnableConcurrentExecution();
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
 
-            context.RegisterSyntaxNodeAction(AnalyzeIdentifiers, SyntaxKind.IdentifierName);
+            // Register actions for handling both C# and VB identifiers
+            context.RegisterSyntaxNodeAction(AnalyzeCSharpIdentifier, CS.SyntaxKind.IdentifierName);
+            context.RegisterSyntaxNodeAction(AnalyzeVBIdentifier, VB.SyntaxKind.IdentifierName);
         }
 
+        /// <summary>
+        /// Child types should implement this method to create the diagnostic for their specific scenario.
+        /// </summary>
+        /// <param name="location">The location the diagnostic occurs at.</param>
+        /// <param name="properties">Properties (including the name of the new identifier that code fix providers should substitute in) that should be included in the diagnostic.</param>
+        /// <param name="messageArgs">Arguments (the simple name of the identifier to be replaced and the full name of the identifier to replace it) to be used in diagnotic messages.</param>
+        /// <returns>A diagnostic to be shown to the user.</returns>
         protected abstract Diagnostic CreateDiagnostic(Location location, ImmutableDictionary<string, string?> properties, params object[] messageArgs);
 
-        private void AnalyzeIdentifiers(SyntaxNodeAnalysisContext context)
+        private void AnalyzeCSharpIdentifier(SyntaxNodeAnalysisContext context)
         {
-            var identifier = (IdentifierNameSyntax)context.Node;
-            var name = identifier.Identifier.ValueText;
+            var identifier = (CSSyntax.IdentifierNameSyntax)context.Node;
+            AnalyzeIdentifier(context, identifier.Identifier.ValueText);
+        }
 
-            // If the node isn't an identifier, bail out
-            if (name is null)
-            {
-                return;
-            }
+        private void AnalyzeVBIdentifier(SyntaxNodeAnalysisContext context)
+        {
+            var identifier = (VBSyntax.IdentifierNameSyntax)context.Node;
+            AnalyzeIdentifier(context, identifier.Identifier.ValueText);
```

Comment:
They were separate because the `Identifier` property is defined separately on VB and C# identifier syntax types. Looking at it now, though, I can just get the identifier name from an extension method and register a single SyntaxNodeAction. I'll do that.

Reply:
Actually, I still need to register to syntax node actions because those are language-specific, but I can still get the simple name in an extension method so that we can get rid of these methods.